### PR TITLE
feat(IDX): allow tests to depend on mainnet pocket-ic

### PR DESCRIPTION
**Pull request by nmattia**
This adds a build setting & related transition that allows for the pocket-ic version of tests to be overriden. By default `//:pocket-ic-server` points to the pocket-ic server from the source tree. This can be overriden by using `pocket_ic_mainnet_test` or by using the `//:pocket-ic-mainnet` target.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -98,3 +99,54 @@ test_suite(
     tags = ["manual"],
     tests = ["//rs/tests/testing_verification/testnets:single_large_node"],
 )
+
+### Pocket IC
+
+# The pocket-ic server binary. Use this as a test dependency if the test
+# does not require a specific pocket-ic version (see ":pocket-ic-variant"
+# for details). Use this
```
Comment (nmattia) [inline BUILD.bazel:107]:
```suggestion
# for details).
```

File: BUILD.bazel [hunk 2]
```diff
@@ -98,3 +99,54 @@ test_suite(
...
+# A "mainnet" variant of the pocket-ic server which represents a
+# released version of pocket-ic.
+config_setting(
+    name = "pocket-ic-variant-mainnet",
+    flag_values = {
+        ":pocket-ic-variant": "mainnet",
+    },
+)
+
+# The pocket-ic as released; use this for tests that ensure consistency
```
Comment (nmattia) [inline BUILD.bazel:145]:
```suggestion
# The pocket-ic as released; use this for tests that need to ensure consistency
```

File: rs/bitcoin/ckbtc/kyt/BUILD.bazel [hunk 3]
```diff
@@ -72,3 +73,8 @@ rust_ic_test(
         "@crate_index//:candid",
     ],
 )
+
+pocket_ic_mainnet_test(
```
Comment (nmattia) [inline rs/bitcoin/ckbtc/kyt/BUILD.bazel:77]: this is just an example, not sure if meaningful or not

**Review by cgundy — APPROVED** [review]
Rubber stamp

**Review by ninegua — APPROVED** [review]

File: BUILD.bazel [hunk 4]
```diff
@@ -98,3 +99,54 @@ test_suite(
...
+        ":pocket-ic-variant-head": "//rs/pocket_ic_server:pocket-ic-server",
+        "//conditions:default": "//rs/pocket_ic_server:pocket-ic-server",
+    }),
+)
+
+# A setting to switch between different variants of pocket-ic. The
+# default pocket-ic variant/version (head) is the one as in the
+# source tree.
+string_setting(
+    name = "pocket-ic-variant",
```
Comment (basvandijk) [inline BUILD.bazel:124]: How about being explicit about which pocket-ic "thing" we're mentioning here? pocket-ic usually refers to the library while the following is about the server:
```suggestion
    name = "pocket-ic-server-variant",
```
Same comment for the other things below.

Comment (nmattia) [reply BUILD.bazel:124]: yep that makes sense!

File: WORKSPACE.bazel [hunk 5]
```diff
@@ -981,3 +981,9 @@ http_file(
     sha256 = "2a428e8d35518ce22002e498f7d618a9eeeddf57371f965e92cf480dd3cbd046",
     url = "https://github.com/dfinity/examples/releases/download/rust-basic-bitcoin-24-09-16/basic_bitcoin.wasm.gz",
 )
+
+http_file(
+    name = "pocket-ic-mainnet-gz",
+    sha256 = "454891cac2421f3f894759ec5e6b6e48fbb544d79197bc29b88d34b93d78a4f1",
+    url = "https://download.dfinity.systems/ic/52ebccfba8855e23dcad9657a8d6e6be01df71f9/binaries/x86_64-linux/pocket-ic.gz",
```
Comment (basvandijk) [inline WORKSPACE.bazel:988]: I see mainnet is using `c66489953ed34670667ad81ce42a172948f289f5`.

Will you address syncing this with [testnet/mainnet_revisions.json#L3](https://github.com/dfinity/ic/blob/master/testnet/mainnet_revisions.json#L3) in another PR?

Comment (nmattia) [reply WORKSPACE.bazel:988]: > Will you address syncing this with [testnet/mainnet_revisions.json#L3](https://github.com/dfinity/ic/blob/master/testnet/mainnet_revisions.json#L3) in another PR?

Yep, talked to DRE team and we'll be taking over the update of that file too. Let's sync offline

**Pull request merged**